packages/curve-ui-kit/src/features/user-profile/UserProfileButton.tsx

```diff
@@ -0,0 +1,19 @@
+import IconButton from '@mui/material/IconButton'
+import PersonIcon from '@mui/icons-material/Person'
+import { useSwitch } from '@ui-kit/hooks/useSwitch'
+
+import { UserProfile } from './UserProfile'
+
+export const UserProfileButton = () => {
+  const [isOpen, open, close] = useSwitch()
+
+  return (
+    <>
+      <IconButton size="small" onClick={open}>
+        <PersonIcon />
+      </IconButton>
+
+      <UserProfile open={isOpen ?? false} onClose={close} />
```

Comment:
This is fine, but if you want false to be the default, it could be passed as arg to `useSwitch`

Reply:
Should we update `useSwitch` such that it's strictly a boolean and the type of `const isOpen: boolean | undefined` becomes `const isOpen: boolean`?

Reply:
I want it to be undefined in the places where I used it, meaning it hasn't been initialized yet